### PR TITLE
Verify container metadata is synced on sharded containers

### PR DESCRIPTION
**Pull request by thiagodasilva**
Added a bit more testing to make sure metadata is synced as expected
when syncing a sharded container

### CODE REVIEW

**Comment by thiagodasilva** [comment]
will need to adjust `requirements.txt` once new container_crawler is released.

File: test/integration/test_sync_sharded_container.py [hunk 1]
```diff
@@ -55,12 +55,14 @@ def _get_broker(self, part, node, account=None, container=None):
...
 
-    def assert_container_object_count(self, conn, cont, expected_obj_count):
+    def assert_container_metadata(self, conn, cont,
+                                  expected_obj_count, expected_user_meta=None):
         headers = conn.head_container(cont)
         self.assertIn('x-container-object-count', headers)
         self.assertEqual(str(expected_obj_count),
                          headers['x-container-object-count'])
 
+
```
Comment (thiagodasilva) [inline test/integration/test_sync_sharded_container.py]: duh...forgot to actually add test :man_facepalming:

File: test/integration/test_sync_sharded_container.py [hunk 2]
```diff
@@ -55,12 +55,14 @@ def _get_broker(self, part, node, account=None, container=None):
         db_file = os.path.join(container_dir, container_hash + '.db')
         return ContainerBroker(db_file)
 
-    def assert_container_object_count(self, conn, cont, expected_obj_count):
+    def assert_container_metadata(self, conn, cont,
+                                  expected_obj_count, expected_user_meta=None):
```
Comment (mdegerne) [inline test/integration/test_sync_sharded_container.py]: You never actually check the expected_user_meta

**Comment by mdegerne** [comment]
Don't merge before the container_crawler reference is fixed, though.

**Comment by timuralp** [comment]
@thiagodasilva this looks good. Can you fix up the missing test and the container crawler pointer? I pushed a new version of the container crawler with the required change, tagged as 0.1.3.

**Comment by thiagodasilva** [comment]
@timuralp thanks for the review. The test was added in this commit: https://github.com/swiftstack/1space/pull/245/commits/d6793c09e327dba96d4ebbc6ae2d6d146e1d4b9e

I'll update the crawler pointer before merging.

**Comment by timuralp** [comment]
Fixed up `requirements.txt` to point to ContainerCrawler 0.1.3 and merged via c6704509a6e00f8aa55ddd5ddea4aeaba46b607f.

**Pull request closed without merging**